### ISSUE TITLE
Optional warning when connecting a node that's already in a 1:1 relationship

### ISSUE THREAD

**Issue by ejoebstl** [prisma/prisma#152]
**Schema:**

```
type File {
  name: String!
}
type User {
  id: ID!
  file: File!
}
```

Where User.file is a 1:1 relation. 

**Use-Case:** 

calling `UpdateUser` with a `fileId` of a File that is already connected to another user silently disconnects the other user. 

**Expected result:** 

In my opinion, this should fail with an error.

**Comment by marktani** [comment]
Thanks for the suggestion!

This would mean that in case a file is already connected, you would need to do an extra query first to check if the file is already related, and then two mutations except for one.

On the other hand, it gives you more control/transparency about what's actually happening.

**Comment by stlbucket** [comment]
there should at least be a warning.  or maybe a property on the relationship that would allow a choice between silent, warning, and error

**Comment by marktani** [comment]
I really like the idea of configuring this behavior, thanks for bringing this up!

**Comment by laxgoalie392** [comment]
My 2 cents are that is should throw an error (or be configurable at the very least). Adding a relationship like this implies that their is a constraint on the data.

What brought me here was that a `Create*` can disconnect an existing relationship. In this case I would expect an error where I can report that this relation already exists and cannot be undone (until the first relation is deleted).

**Comment by marktani** [comment]
This is probably solved when we introduced #41. The configuration would be done by setting a relation as required or not.

Or do you think that's a different concept and the warning should be configurable even for non-required relations?

**Comment by laxgoalie392** [comment]
@marktani this was actually the first thing i looked for! I definitely think this would help and fit in with existing functionality
